extmod/network_lwip.c

```diff
@@ -90,6 +100,241 @@ mp_obj_t mod_network_nic_ifconfig(struct netif *netif, size_t n_args, const mp_o
     }
 }
 
+mp_obj_t network_nic_config(struct netif *netif, size_t n_args, const mp_obj_t *args, mp_map_t *kwargs) {
+
+    if (kwargs->used == 0) {
+        // Get config value
+        if (n_args != 1) {
+            mp_raise_TypeError(MP_ERROR_TEXT("must query one param"));
+        }
+
+        switch (mp_obj_str_get_qstr(args[0])) {
+            case MP_QSTR_dhcp4: {
+                struct dhcp *dhcp = netif_dhcp_data(netif);
+                return dhcp != NULL && dhcp->state != DHCP_STATE_OFF ? mp_const_true : mp_const_false;
+            }
+            #if LWIP_IPV6_DHCP6
+            case MP_QSTR_dhcp6: {
+                struct dhcp6 *dhcp = netif_dhcp6_data(netif);
+                return dhcp != NULL && dhcp->state != DHCP6_STATE_OFF ? mp_const_true : mp_const_false;
+            }
+            #endif
+            #if LWIP_IPV6_AUTOCONFIG
+            case MP_QSTR_autoconf6: {
+                return netif->ip6_autoconfig_enabled ?  mp_const_true : mp_const_false;
+            }
+            #endif
+            case MP_QSTR_dns: {
+                const char *addr_str = ipaddr_ntoa(dns_getserver(0));
```

Comment:
See comment later about `ipaddr_ntoa` (would be good to use `ipaddr_ntoa_r` instead).